pkg/sync/grpc/grpc_sync.go

```diff
@@ -0,0 +1,193 @@
+package grpc
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"math"
+	"strings"
+	"time"
+
+	"google.golang.org/grpc/credentials/insecure"
+
+	"buf.build/gen/go/open-feature/flagd/grpc/go/sync/v1/syncv1grpc"
+	v1 "buf.build/gen/go/open-feature/flagd/protocolbuffers/go/sync/v1"
+
+	"github.com/open-feature/flagd/pkg/logger"
+	"github.com/open-feature/flagd/pkg/sync"
+	"golang.org/x/sync/errgroup"
+	"google.golang.org/grpc"
+)
+
+const (
+	// Prefix for GRPC URL inputs. GRPC does not define a prefix through standard. This prefix helps to differentiate
+	// remote URLs for REST APIs (i.e - HTTP) from GRPC endpoints.
+	Prefix = "grpc://"
+
+	// Connection retry constants
+	// Backoff period is calculated with backOffBase ^ #retry-iteration. However, when backoffLimit is reached, fallback
+	// to constantBackoffDelay
+	backoffLimit         = 3
+	backOffBase          = 4
+	constantBackoffDelay = 60
+)
+
+type Sync struct {
+	Target     string
+	ProviderID string
+	Logger     *logger.Logger
+}
+
+func (g *Sync) Sync(ctx context.Context, dataSync chan<- sync.DataSync) error {
+	options := []grpc.DialOption{
+		grpc.WithTransportCredentials(insecure.NewCredentials()),
+	}
+
+	// initial dial and connection. Failure here must result in a startup failure
+	dial, err := grpc.DialContext(ctx, g.Target, options...)
+	if err != nil {
+		g.Logger.Error(fmt.Sprintf("Error establishing grpc connection: %s", err.Error()))
+		return err
+	}
+
+	serviceClient := syncv1grpc.NewFlagSyncServiceClient(dial)
+	syncClient, err := serviceClient.SyncFlags(context.Background(), &v1.SyncFlagsRequest{ProviderId: g.ProviderID})
```

Comment:
```suggestion
	syncClient, err := serviceClient.SyncFlags(ctx, &v1.SyncFlagsRequest{ProviderId: g.ProviderID})
```
Adding suggestions which will allow for graceful shutdown, also requires the below suggestions

Reply:
Good one, this slipped through somehow  😞

Reply:
Updated with the latest commit